docs/technical-overview/architecture.md

```diff
@@ -35,6 +35,20 @@ rule, you should avoid relying on JavaScript for layout when working on DEV.
 We use [PreactJS][preact], a lightweight alternative to ReactJS, and we try to
 reduce our bundle size with [dynamic imports][dynamic_imports].
 
+## Service workers and shell architecture
+
+We make use of serviceworkers to cache portions of the page.
+
+We cache styles here as well as script fingerprinting, so we should increment
+the number in `/async_info/shell_version` any time we change core CSS or
+JavaScript.
+
+Serviceworkers can be controlled in the `application` tab of Chrome.
+Serviceworkers are a reverse proxy that runs in the browser in a non-blocking
+thread, supported by most major browsers. You may want to disable or bypass
```

Comment:
what if we add an env var that can help deactivating the service worker on dev?

Reply:
I'm down to try. I can think of some gotcha scenarios if not done just right.